coderd/database/pubsub/pubsub.go

```diff
@@ -541,18 +544,16 @@ func (p *PGPubsub) startListener(ctx context.Context, connectURL string) error {
 				p.logger.Error(ctx, "pubsub failed to connect to postgres", slog.Error(err))
 			}
 			// This callback gets events whenever the connection state changes.
-			// Don't send if the errChannel has already been closed.
-			select {
-			case <-errCh:
+			// Only send the first error.
+			if sentErrCh {
 				return
-			default:
-				errCh <- err
-				close(errCh)
 			}
+			errCh <- err // won't block because we are buffered.
+			sentErrCh = true
```

Comment:
Do we need to protect this or is sequential execution guaranteed?

Reply:
It executes sequentially, at least by inspection. There is always a chance they could change the implementation and call-back concurrently, but it didn't seem worth while defending for that.